vulture_os/authentication/ldap/tools.py

```diff
@@ -127,12 +127,14 @@ def create_group(ldap_repository, group_name, members_username):
 
 
 def create_user(ldap_repository, group_name, user_name, userPassword, attrs):
-    group_dn = group_name
+    group_dn = False
+    if group_name:
+        group_dn = group_name
 
-    if ldap_repository.base_dn not in group_dn:
-        group_dn = ldap_repository.create_group_dn(group_name)
+        if ldap_repository.base_dn not in group_dn:
+            group_dn = ldap_repository.create_group_dn(group_name)
 
-    user_dn = f"{ldap_repository.user_attr}={user_name},{group_dn}"
+    user_dn = ldap_repository.create_user_dn(user_name)
```

Comment:
is 'group_dn' not used anymore for 'user_dn' generation on purpose ? It doesn't seem to be used in create_user_dn() so the resulting user_dn might be different

Reply:
The group_dn must not be in the user_dn. In the case where the group does not exist, we must be able to create user anyway.